tests/test_caching.py

```diff
@@ -207,9 +208,11 @@ def test_restore_cache_prefix_timestamps(runner, tmp_dir_name):
     test_files_oldest = setup_cache_test_files(tmp_dir_name=tmp_dir_name,
                                                cache_name=cache_name_oldest,
                                                num_files=3)
+    sleep(.2)
```

Comment:
Sleeps?

Reply:
Yeah this was super lame! This test was failing in the first few builds in this PR. Nothing changed in the caching and I theorized that it was a timing issue. This test is sensitive to the timestamps of the caches so adding the .2 seconds sleep just ensure the timestamp difference was large enough so that when the caching scheme compares timestamps it picks the one that was created last. I didn't dig into this to know for 100% certain but I believe the test files were created so fast on the server that the timestamps weren't different enough to detect which one was more recent.

FYI: The test was failing because it was choosing the cache of files created in the middle, rather than the one most recently.